frontend/src/templates/components/email.js

```diff
@@ -49,24 +49,31 @@ class Email extends Component {
 
         api.post(API_DOMAIN + 'announcement/' + this.props.source.toLowerCase() + id_param_string, data)
         .then(res => {
-            self.setState({ message_sent: res.data.success ? 1 : -1 })
-            self.bodyField.value = ''
-            self.subjectField.value = ''
-            self.setState({ subject: ''})
-            self.setState({ body: ''})
+            this.setState({ message_sent: res.data.success ? 1 : -1 })
+            this.bodyField.value = ''
+            this.subjectField.value = ''
+            this.setState({ subject: ''})
+            this.setState({ body: ''})
     
         }).catch (function(error) {
             if (error.response.status !== 200) {
-                self.setState({ error_status: true });
-                self.setState({ error_code: error.response.status });
-                self.setState({ message_sent: -1 })
+                this.setState({ error_status: true });
+                this.setState({ error_code: error.response.status });
+                this.setState({ message_sent: -1 })
             }
         } 
         )
     }
 
     componentDidMount() {
-
+        console.log(this.props.source.toLowerCase() + "/detail?id=" + this.props.params.id)
+        api.get(`${this.props.source.toLowerCase()}/detail?id=${this.props.params.id}`).then(res => {
+            console.log(res)
+            const details = res.data;
+            console.log(details)
```

Comment:
Might want to delete the console.logs once we can verify that theyre fully functional